parser/match_operations.go

```diff
@@ -437,21 +437,48 @@ func comparisonMatch(
 		if err := matchIndexValid(matches, amountMatch[1]); err != nil {
 			return fmt.Errorf("%w: opposite amounts comparison error", err)
 		}
-		for _, op := range matches[amountMatch[0]].Operations {
-			for _, otherOp := range matches[amountMatch[1]].Operations {
-				if err := oppositeAmounts(
-					op,
-					otherOp,
-				); err != nil {
-					return fmt.Errorf("%w: amounts not opposites", err)
-				}
-			}
+		if err := validateMatchOperationsForOppositeComparison(amountMatch[0], matches); err != nil {
+			return err
+		}
+		if err := validateMatchOperationsForOppositeComparison(amountMatch[1], matches); err != nil {
+			return err
+		}
+		// check for opposites amount
```

Comment:
nit[STYLE]: should be enter above this line

nit: can we write a clearer comment here?